corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -13,87 +13,57 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
+                return "";
             }
-            return value;
         },
         decodeValue = function (model, value) {
-            if (!value) {
-                return value;
-            }
-            if (model.get("input") === "daterange") {
-                return value.replace("__range__", "").replace("__", separator);
+            if (!_.isString(value)) {
+                return [false, undefined];
             }
-            if (model.get('input') === 'select') {
-                return value.split(selectDelimiter);
-            }
-            return value;
-        };
-
-    var QueryView = Marionette.View.extend({
-        tagName: "tr",
-        className: "formplayer-request",
-        template: _.template($("#query-view-item-template").html() || ""),
+            var values = value.split(selectDelimiter),
+                searchForBlank = _.contains(values, ""),
+                values = _.without(values, "");
 
-        templateContext: function () {
-            var imageUri = this.options.model.get('imageUri'),
-                audioUri = this.options.model.get('audioUri'),
-                appId = this.model.collection.appId,
-                value = this.options.model.get('value');
-
-            return {
-                imageUrl: imageUri ? FormplayerFrontend.getChannel().request('resourceMap', imageUri, appId) : "",
-                audioUrl: audioUri ? FormplayerFrontend.getChannel().request('resourceMap', audioUri, appId) : "",
-                value: value,
-            };
-        },
-
-        initialize: function () {
-            // If input doesn't have a default value, check to see if there's a sticky value from user's last search
-            if (!this.options.model.get('value')) {
-                var stickyValue = hqImport("cloudcare/js/formplayer/utils/util").getStickyQueryInputs()[this.options.model.get('id')];
-                this.options.model.set('value', decodeValue(this.options.model, stickyValue));
+            if (model.get('input') === 'select') {
+                value = values;
+            } else if (values.length === 1) {
+                value = values[0];
+                if (model.get("input") === "daterange") {
+                    value = value.replace("__range__", "").replace("__", separator);
+                }
+            } else {
+                value = undefined;
             }
+            return [searchForBlank, value];
         },
-
-        ui: {
-            valueDropdown: 'select.query-field',
-            hqHelp: '.hq-help',
-            dateRange: 'input.daterange',
-            queryField: '.query-field',
-        },
-
-        modelEvents: {
-            'change': 'render',
-        },
-
-        geocoderItemCallback: function (addressTopic) {
+        geocoderItemCallback = function (addressTopic, model) {
             return function (item) {
+                model.set('value', item.place_name);
+                initMapboxWidget(model);
```

Comment:
@esoergel do you recall why it was necessary to call `initMapboxWidget`?

This function handles events from the widget so I don't see how calling `initMapboxWidget` make sense.

Reply:
I don't recall, but I agree it looks pretty weird.  It was tricky to manage control flow - one guess is maybe something cleared out the input in the widget and re-initializing the widget set it back?